src/box/relay.cc

```diff
@@ -636,12 +637,25 @@ tx_status_update(struct cmsg *msg)
 
 /**
  * Update replica gc state in tx thread.
+ * Note that there is a case when replica and its consumer were dropped
+ * from system spaces, but the relay is not stopped yet - we must not create
+ * new gc consumer in this case, because it will become hanging. So the
+ * function tries to update gc_consumer - it is not created if it doesn't
+ * exist.
  */
 static void
 tx_gc_advance(struct cmsg *msg)
 {
 	struct relay_gc_msg *m = (struct relay_gc_msg *)msg;
-	gc_consumer_advance(m->relay->replica->gc, &m->vclock);
+	struct credentials *old_user = effective_user();
+
+	fiber_set_user(fiber(), &admin_credentials);
+	if (box_gc_consumer_update(m->relay->replica, &m->vclock) != 0) {
```

Comment:
If user manually drops consumer from `_gc_consumers` and then tries to resubscribe, the consumer should be recreated. Otherwise, it'll be an error during resubscribe.

And probably, we should prohibit deleting row from `gc_consumers`, while there's a row in `_cluster`

Reply:
We set consumer in `relay_subscribe`, so it will be re-created on resubscribe.

Reply:
> And probably, we should prohibit deleting row from gc_consumers, while there's a row in _cluster

Hm, I'll think about it.

Reply:
> And probably, we should prohibit deleting row from gc_consumers, while there's a row in _cluster

Then consumers of anonymous replicas won't work.